packages/kotti-ui/source/kotti-field-select/components/GenericSelectField.vue

```diff
@@ -0,0 +1,379 @@
+			if (field.currentValue === null) return null
+			return (
+				props.options.find((option) => option.value === field.currentValue)
+					?.label ?? null
+			)
+		})
+
+		const { forceUpdateKey, forceUpdate } = useForceUpdate()
+
+		watch(isDropdownMounted, (isMounted) => {
+			if (isMounted) return
+
+			if (props.isRemote) {
+				if (props.query !== null) emit(UPDATE_QUERY, null)
+			} else localQuery.value = null
+		})
+
+		watch(isDropdownOpen, (isOpen) => {
+			if (!isOpen) return
+
+			inputRef.value?.focus()
+		})
+
+		const queryValue = computed(() =>
+			props.isRemote ? props.query : localQuery.value,
+		)
+
+		/**
+		 * keeps reference of already seen option labels so that
+		 * the tags can be rendered properly when filtering remotely
+		 */
+		const seenValueLabelMap = ref<Map<Shared.Value, string>>(new Map())
+
+		watch(
+			() => props.options,
+			() => {
+				seenValueLabelMap.value = new Map([
+					...seenValueLabelMap.value,
+					...props.options.map((option): [Shared.Value, string] => [
+						option.value,
+						option.label,
+					]),
+				])
+			},
+			{ immediate: true },
+		)
+
+		return {
+			collapsedTagCount: computed(() =>
+				props.isMultiple
+					? (field.currentValue as MultiValue).length - props.collapseTagsAfter
+					: 0,
+			),
+			field,
+			filteredOptions: computed(() =>
+				props.isRemote
+					? props.options
+					: props.options.filter(
+							(o) =>
+								localQuery.value === null ||
+								o.label.toLowerCase().includes(localQuery.value.toLowerCase()),
+					  ),
+			),
+			inputProps: computed(() => ({
+				...field.inputProps,
+				class: ['kt-field-select__wrapper'],
+				forceUpdateKey: forceUpdateKey.value,
+				placeholder: props.placeholder ?? undefined,
+				size: 1,
+				type: 'text',
+				value: (() => {
+					if (isDropdownOpen.value) return queryValue.value ?? undefined
+
+					return selectedLabel.value ?? undefined
+				})(),
+			})),
+			inputRef,
+			isDropdownOpen,
+			isInputVisible: computed(
+				// save some space in multi select when tags are shown
+				() => !props.isMultiple || isDropdownOpen.value || field.isEmpty,
+			),
+			tippyContentRef: selectTippy.tippyContentRef,
+			tippyTriggerRef: selectTippy.tippyTriggerRef,
+			onOptionsInput: (value: MultiValue) => {
+				field.setValue(props.isMultiple ? value : value?.[0] ?? null)
+
+				inputRef.value?.focus()
+				// single select: close the tippy instance whenever a selection is made.
+				// This (watcher on isDropdownMounted) also intentionally resets the query
+				// so that the api-call (for example) can already trigger and load the non-filtered options
+				if (!props.isMultiple) selectTippy.setIsDropdownOpen(false)
+			},
+			onPressDelete: () => {
+				if (queryValue.value || !props.isMultiple) return // only delete value if query is already empty
+
+				const value = field.currentValue as MultiValue
+				if (value.length > 0) field.setValue(value.slice(0, -1))
```

Comment:
argument from Florian: if the single select is focused and dropdown is open, and we delete the selected value upon keydown.delete then the user won't know, because the query is shown and not the "value"